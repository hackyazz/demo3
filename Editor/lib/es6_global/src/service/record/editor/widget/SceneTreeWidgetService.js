


function getWidget(param) {
  return /* SceneTree */0;
}

function isWidget(startWidget) {
  if (startWidget !== undefined) {
    return startWidget === /* SceneTree */0;
  } else {
    return false;
  }
}

export {
  getWidget ,
  isWidget ,
  
}
/* No side effect */
