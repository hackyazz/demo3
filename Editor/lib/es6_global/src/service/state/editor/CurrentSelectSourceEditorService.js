


function getCurrentSelectSource(editorState) {
  return editorState[/* currentSelectSource */12];
}

function setCurrentSelectSource(currentSelectSource, editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */currentSelectSource,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

function clearCurrentSelectSource(editorState) {
  return /* record */[
          /* settingRecord */editorState[/* settingRecord */0],
          /* sceneTreeRecord */editorState[/* sceneTreeRecord */1],
          /* assetRecord */editorState[/* assetRecord */2],
          /* sceneViewRecord */editorState[/* sceneViewRecord */3],
          /* gameViewRecord */editorState[/* gameViewRecord */4],
          /* eventRecord */editorState[/* eventRecord */5],
          /* imguiRecord */editorState[/* imguiRecord */6],
          /* inspectorRecord */editorState[/* inspectorRecord */7],
          /* consoleRecord */editorState[/* consoleRecord */8],
          /* transformRecord */editorState[/* transformRecord */9],
          /* pickingRecord */editorState[/* pickingRecord */10],
          /* currentDragSource */editorState[/* currentDragSource */11],
          /* currentSelectSource */undefined,
          /* loopId */editorState[/* loopId */13],
          /* languageType */editorState[/* languageType */14]
        ];
}

export {
  getCurrentSelectSource ,
  setCurrentSelectSource ,
  clearCurrentSelectSource ,
  
}
/* No side effect */
