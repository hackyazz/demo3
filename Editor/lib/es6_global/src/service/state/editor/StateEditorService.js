

import * as CreateEditorStateDataEditorService$WonderEditor from "../data/CreateEditorStateDataEditorService.js";

function getStateIsDebug(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1];
}

function setStateIsDebug(isDebug) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isDebug */1] = isDebug;
  return /* () */0;
}

function getIsRun(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2];
}

function setIsRun(isRun) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* isRun */2] = isRun;
  return /* () */0;
}

function getState(param) {
  return CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0];
}

function setState(state) {
  CreateEditorStateDataEditorService$WonderEditor.editorStateData[/* editorState */0] = state;
  return state;
}

export {
  getStateIsDebug ,
  setStateIsDebug ,
  getIsRun ,
  setIsRun ,
  getState ,
  setState ,
  
}
/* CreateEditorStateDataEditorService-WonderEditor Not a pure module */
