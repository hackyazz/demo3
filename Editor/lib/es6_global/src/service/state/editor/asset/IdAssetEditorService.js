

import * as IdAssetService$WonderEditor from "../../../record/editor/asset/IdAssetService.js";
import * as IndexAssetEditorService$WonderEditor from "./IndexAssetEditorService.js";

function generateNodeId(editorState) {
  var match = IdAssetService$WonderEditor.generateNodeId(IndexAssetEditorService$WonderEditor.getNodeIndex(editorState));
  return /* tuple */[
          IndexAssetEditorService$WonderEditor.setNodeIndex(match[0], editorState),
          match[1]
        ];
}

export {
  generateNodeId ,
  
}
/* No side effect */
