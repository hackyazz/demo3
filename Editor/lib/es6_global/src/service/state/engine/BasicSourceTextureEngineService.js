

import * as BasicSourceTextureAPI$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/api/texture/BasicSourceTextureAPI.js";
import * as BufferSourceTextureService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/texture/BufferSourceTextureService.js";
import * as InitSourceTextureMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/texture/InitSourceTextureMainService.js";
import * as NameBasicSourceTextureMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/texture/basic_source/NameBasicSourceTextureMainService.js";
import * as OperateBasicSourceTextureMainService$Wonderjs from "../../../../../../node_modules/wonder.js/lib/es6_global/src/service/state/main/texture/basic_source/OperateBasicSourceTextureMainService.js";

function setSource(source, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureSource(texture, source, engineState);
}

function setBasicSourceTextureName(name, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureName(texture, name, engineState);
}

function setWrapS(wrapS, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureWrapS(texture, wrapS, engineState);
}

function setWrapT(wrapT, texture, engineTtate) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureWrapT(texture, wrapT, engineTtate);
}

function setMagFilter(filter, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureMagFilter(texture, filter, engineState);
}

function setMinFilter(filter, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureMinFilter(texture, filter, engineState);
}

function setFormat(format, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureFormat(texture, format, engineState);
}

function setType(type_, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureType(texture, type_, engineState);
}

function setFlipY(filpY, texture, engineState) {
  return BasicSourceTextureAPI$Wonderjs.setBasicSourceTextureFlipY(texture, filpY, engineState);
}

var getIsNeedUpdate = OperateBasicSourceTextureMainService$Wonderjs.getIsNeedUpdate;

function setIsNeedUpdate(isNeedUpdate, texture, state) {
  return OperateBasicSourceTextureMainService$Wonderjs.setIsNeedUpdate(texture, isNeedUpdate ? BufferSourceTextureService$Wonderjs.getNeedUpdate(/* () */0) : BufferSourceTextureService$Wonderjs.getNotNeedUpdate(/* () */0), state);
}

function initTexture(texture, state) {
  return InitSourceTextureMainService$Wonderjs.initTexture(texture, state);
}

var create = BasicSourceTextureAPI$Wonderjs.createBasicSourceTexture;

var unsafeGetSource = BasicSourceTextureAPI$Wonderjs.unsafeGetBasicSourceTextureSource;

var getBasicSourceTextureName = NameBasicSourceTextureMainService$Wonderjs.getName;

var unsafeGetBasicSourceTextureName = BasicSourceTextureAPI$Wonderjs.unsafeGetBasicSourceTextureName;

var getWidth = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureWidth;

var getHeight = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureHeight;

var getWrapS = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureWrapS;

var getWrapT = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureWrapT;

var getMagFilter = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureMagFilter;

var getMinFilter = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureMinFilter;

var getFormat = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureFormat;

var getType = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureType;

var getFlipY = BasicSourceTextureAPI$Wonderjs.getBasicSourceTextureFlipY;

var getAllTextures = BasicSourceTextureAPI$Wonderjs.getAllTextures;

export {
  create ,
  unsafeGetSource ,
  setSource ,
  getBasicSourceTextureName ,
  unsafeGetBasicSourceTextureName ,
  setBasicSourceTextureName ,
  getWidth ,
  getHeight ,
  getWrapS ,
  setWrapS ,
  getWrapT ,
  setWrapT ,
  getMagFilter ,
  setMagFilter ,
  getMinFilter ,
  setMinFilter ,
  getFormat ,
  setFormat ,
  getType ,
  setType ,
  getFlipY ,
  setFlipY ,
  getIsNeedUpdate ,
  setIsNeedUpdate ,
  initTexture ,
  getAllTextures ,
  
}
/* BasicSourceTextureAPI-Wonderjs Not a pure module */
