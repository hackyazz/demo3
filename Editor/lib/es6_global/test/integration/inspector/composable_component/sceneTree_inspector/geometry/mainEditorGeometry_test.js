

import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as WDBTool$WonderEditor from "../../../../../tool/WDBTool.js";
import * as LoadTool$WonderEditor from "../../../../asset/tool/LoadTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as GameObjectTool$WonderEditor from "../../../../../tool/GameObjectTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../tool/BuildComponentTool.js";
import * as GeometryToolEngine$WonderEditor from "../../../../../tool/engine/GeometryToolEngine.js";
import * as StateEditorService$WonderEditor from "../../../../../../src/service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorAssetTool$WonderEditor from "../../../../asset/tool/MainEditorAssetTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as IdAssetEditorService$WonderEditor from "../../../../../../src/service/state/editor/asset/IdAssetEditorService.js";
import * as GeometryEngineService$WonderEditor from "../../../../../../src/service/state/engine/GeometryEngineService.js";
import * as MainEditorGeometryTool$WonderEditor from "./tool/MainEditorGeometryTool.js";
import * as SceneTreeWidgetService$WonderEditor from "../../../../../../src/service/record/editor/widget/SceneTreeWidgetService.js";
import * as MainEditorAssetTreeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetTreeTool.js";
import * as MainEditorAssetUploadTool$WonderEditor from "../../../../asset/tool/MainEditorAssetUploadTool.js";
import * as LightMaterialEngineService$WonderEditor from "../../../../../../src/service/state/engine/LightMaterialEngineService.js";
import * as MainEditorAssetWDBNodeTool$WonderEditor from "../../../../asset/tool/MainEditorAssetWDBNodeTool.js";
import * as BasicSourceTextureEngineService$WonderEditor from "../../../../../../src/service/state/engine/BasicSourceTextureEngineService.js";
import * as CurrentSelectSourceEditorService$WonderEditor from "../../../../../../src/service/state/editor/CurrentSelectSourceEditorService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../src/service/state/engine/GameObjectComponentEngineService.js";

describe("MainEditorGeometry component", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initState(sandbox, undefined, undefined, /* () */0);
                MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode);
                var partial_arg = SceneTreeWidgetService$WonderEditor.getWidget(/* () */0);
                return StateLogicService$WonderEditor.getAndSetEditorState((function (param) {
                              return CurrentSelectSourceEditorService$WonderEditor.setCurrentSelectSource(partial_arg, param);
                            }));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test set currentSceneTreeNode", (function () {
                describe("test change geometry", (function () {
                        describe("test snapshot", (function () {
                                describe("test show select geometry group widget", (function () {
                                        var _addNoTexCoordGeometryWDBGameObject = function () {
                                          var editorState = StateEditorService$WonderEditor.getState(/* () */0);
                                          var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                          var match = GeometryToolEngine$WonderEditor.createGameObjectAndSetPointData(engineState, false, /* () */0);
                                          var match$1 = IdAssetEditorService$WonderEditor.generateNodeId(editorState);
                                          var editorState$1 = MainEditorAssetWDBNodeTool$WonderEditor.addWDBNodeToRoot(match[1], match$1[1], match$1[0], match[4], undefined, /* () */0);
                                          StateEditorService$WonderEditor.setState(editorState$1);
                                          StateEngineService$WonderEditor.setState(match[0]);
                                          return /* () */0;
                                        };
                                        var _setGameObjectLightMateiralDiffuseMap = function (gameObject) {
                                          var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                          var match = BasicSourceTextureEngineService$WonderEditor.create(engineState);
                                          var engineState$1 = match[0];
                                          var __x = GameObjectComponentEngineService$WonderEditor.unsafeGetLightMaterialComponent(gameObject, engineState$1);
                                          return StateEngineService$WonderEditor.setState(LightMaterialEngineService$WonderEditor.setLightMaterialDiffuseMap(__x, match[1], engineState$1));
                                        };
                                        Wonder_jest.test("if current material has no map, select geometry group should contain geometry which has texCoord or no texCoord", (function () {
                                                _addNoTexCoordGeometryWDBGameObject(/* () */0);
                                                var currentGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildGeometry(currentGameObjectGeometry, undefined, undefined, undefined, true, /* () */0));
                                              }));
                                        return Wonder_jest.test("else, select geometry group should only contain geometry which has texCoord", (function () {
                                                      _addNoTexCoordGeometryWDBGameObject(/* () */0);
                                                      _setGameObjectLightMateiralDiffuseMap(GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0));
                                                      var currentGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                                      return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildGeometry(currentGameObjectGeometry, undefined, undefined, undefined, true, /* () */0));
                                                    }));
                                      }));
                                return Wonder_jest.test("test hide select geometry group widget", (function () {
                                              var currentGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                              return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildGeometry(currentGameObjectGeometry, undefined, undefined, undefined, false, /* () */0));
                                            }));
                              }));
                        describe("test logic", (function () {
                                Wonder_jest.test("test the current gameObject geometry should be Cube", (function () {
                                        var currentGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                        return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                              return GeometryEngineService$WonderEditor.unsafeGetGeometryName(currentGameObjectGeometry, param);
                                                            }))), MainEditorGeometryTool$WonderEditor.getDefaultCubeGeometryName(/* () */0));
                                      }));
                                Wonder_jest.test("change geometry to be Sphere, the current gameObject geometry should be Sphere", (function () {
                                        MainEditorGeometryTool$WonderEditor.changeGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), MainEditorGeometryTool$WonderEditor.getDefaultSphereGeometryComponent(undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                                        var newGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                        return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                              return GeometryEngineService$WonderEditor.unsafeGetGeometryName(newGameObjectGeometry, param);
                                                            }))), MainEditorGeometryTool$WonderEditor.getDefaultSphereGeometryName(/* () */0));
                                      }));
                                return Wonder_jest.test("test add Cube geometry component again and again, currentSceneTreeNode's geometry should be Cube", (function () {
                                              BuildComponentTool$WonderEditor.buildGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), undefined, undefined, undefined, undefined, /* () */0);
                                              MainEditorGeometryTool$WonderEditor.changeGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), MainEditorGeometryTool$WonderEditor.getDefaultSphereGeometryComponent(undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                                              MainEditorGeometryTool$WonderEditor.changeGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), MainEditorGeometryTool$WonderEditor.getDefaultCubeGeometryComponent(undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                                              MainEditorGeometryTool$WonderEditor.changeGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), MainEditorGeometryTool$WonderEditor.getDefaultSphereGeometryComponent(undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                                              MainEditorGeometryTool$WonderEditor.changeGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), MainEditorGeometryTool$WonderEditor.getDefaultCubeGeometryComponent(undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                                              var newGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](GeometryEngineService$WonderEditor.unsafeGetGeometryName(newGameObjectGeometry, StateEngineService$WonderEditor.unsafeGetState(/* () */0))), "Wonder-Default-Cube");
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("test load asset wdb", (function () {
                        var cubeTexturedWDBArrayBuffer = /* record */[/* contents */1];
                        beforeAll((function () {
                                cubeTexturedWDBArrayBuffer[0] = WDBTool$WonderEditor.convertGLBToWDB("CubeTextured");
                                return /* () */0;
                              }));
                        beforeEach((function () {
                                Curry._1(MainEditorAssetTool$WonderEditor.buildFakeFileReader, /* () */0);
                                Curry._1(MainEditorAssetTool$WonderEditor.buildFakeImage, /* () */0);
                                LoadTool$WonderEditor.buildFakeTextDecoder(LoadTool$WonderEditor.convertUint8ArrayToBuffer);
                                LoadTool$WonderEditor.buildFakeURL(sandbox[0]);
                                return LoadTool$WonderEditor.buildFakeLoadImage();
                              }));
                        Wonder_jest.testPromise("test select geometry group widget should show all geometry", (function () {
                                MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                return MainEditorAssetUploadTool$WonderEditor.loadOneWDB(cubeTexturedWDBArrayBuffer[0], undefined, undefined, "CubeTextured", /* () */0).then((function () {
                                              var component = BuildComponentTool$WonderEditor.buildGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), undefined, undefined, undefined, true, /* () */0);
                                              return Promise.resolve(ReactTestTool$WonderEditor.createSnapshotAndMatch(component));
                                            }));
                              }));
                        return Wonder_jest.testPromise("test set new geometry should set into engineState", (function () {
                                      MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                                      var newGeometry = GeometryToolEngine$WonderEditor.getNewGeometry(undefined, /* () */0);
                                      return MainEditorAssetUploadTool$WonderEditor.loadOneWDB(cubeTexturedWDBArrayBuffer[0], undefined, undefined, "CubeTextured", /* () */0).then((function () {
                                                    var oldGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                                    MainEditorGeometryTool$WonderEditor.changeGeometry(oldGameObjectGeometry, newGeometry, undefined, undefined, undefined, /* () */0);
                                                    var newGameObjectGeometry = GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0);
                                                    return Promise.resolve(Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                              return GeometryEngineService$WonderEditor.unsafeGetGeometryName(newGameObjectGeometry, param);
                                                                            }))), MainEditorGeometryTool$WonderEditor.getCubeTexturedGeometryName(/* () */0)));
                                                  }));
                                    }));
                      }));
                return /* () */0;
              }));
        describe("test select geometry group->show order", (function () {
                var truckWDBArrayBuffer = /* record */[/* contents */1];
                beforeAll((function () {
                        truckWDBArrayBuffer[0] = WDBTool$WonderEditor.convertGLBToWDB("CesiumMilkTruck");
                        return /* () */0;
                      }));
                beforeEach((function () {
                        Curry._1(MainEditorAssetTool$WonderEditor.buildFakeFileReader, /* () */0);
                        Curry._1(MainEditorAssetTool$WonderEditor.buildFakeImage, /* () */0);
                        LoadTool$WonderEditor.buildFakeTextDecoder(LoadTool$WonderEditor.convertUint8ArrayToBuffer);
                        LoadTool$WonderEditor.buildFakeURL(sandbox[0]);
                        LoadTool$WonderEditor.buildFakeLoadImage();
                        MainEditorAssetTreeTool$WonderEditor.BuildAssetTree[/* buildEmptyAssetTree */1](/* () */0);
                        return /* () */0;
                      }));
                return Wonder_jest.testPromise("\n        order should be:\n        1)default geometry is in the end;\n        2)sort geometry assets by firstname alphabetically\n        ", (function () {
                              return MainEditorAssetUploadTool$WonderEditor.loadOneWDB(truckWDBArrayBuffer[0], undefined, undefined, undefined, /* () */0).then((function () {
                                            return Promise.resolve(ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildGeometry(GameObjectTool$WonderEditor.getCurrentGameObjectGeometry(/* () */0), undefined, undefined, undefined, true, /* () */0)));
                                          }));
                            }));
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
