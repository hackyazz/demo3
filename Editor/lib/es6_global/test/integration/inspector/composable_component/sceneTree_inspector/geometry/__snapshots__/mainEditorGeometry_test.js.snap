// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorGeometry component test select geometry group->show order 
        order should be:
        1)default geometry is in the end;
        2)sort geometry assets by firstname alphabetically
         1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Geometry
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Cesium_Milk_Truck_0
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Cesium_Milk_Truck_1
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Cesium_Milk_Truck_2
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wheels
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Sphere
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test hide select geometry group widget 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test show select geometry group widget else, select geometry group should only contain geometry which has texCoord 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Geometry
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Sphere
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test change geometry test snapshot test show select geometry group widget if current material has no map, select geometry group should contain geometry which has texCoord or no texCoord 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Geometry
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          geometryNoTexCoord
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Sphere
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;

exports[`MainEditorGeometry component test set currentSceneTreeNode test load asset wdb test select geometry group widget should show all geometry 1`] = `
<article
  className="wonder-inspector-geometry"
>
  <div
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      Geometry
    </div>
    <div
      className="item-content"
    >
      <div
        className="inspector-select"
      >
        <div
          className="select-name"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-img"
          onClick={[Function]}
        >
          <img
            src="./public/img/select.png"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="select-component-content"
  >
    <div
      className="select-component-item"
    >
      <div
        className="select-item-header"
      >
        Geometry
      </div>
      <div
        className="select-item-body"
      >
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Mesh
        </div>
        <div
          className="select-item-content select-item-active"
          onClick={[Function]}
        >
          Wonder-Default-Cube
        </div>
        <div
          className="select-item-content"
          onClick={[Function]}
        >
          Wonder-Default-Sphere
        </div>
      </div>
    </div>
    <div
      className="select-component-bg"
      onClick={[Function]}
    />
  </div>
</article>
`;
